ext/mbstring/libmbfl/filters/mbfilter_utf7imap.c

```diff
@@ -140,16 +140,18 @@ int mbfl_filt_conv_utf7imap_wchar(int c, mbfl_convert_filter *filter)
 		if (n < 0 || n > 63) {
 			if (c == '-') {
 				if (filter->status == 1) { /* "&-" -> "&" */
+					filter->cache = filter->status = 0;
 					CK((*filter->output_function)('&', filter->data));
 				} else if (filter->cache) {
 					/* Base64-encoded section ended abruptly, with partially encoded characters,
 					 * or it could be that it ended on the first half of a surrogate pair */
+					filter->cache = filter->status = 0;
 					CK((*filter->output_function)(MBFL_BAD_INPUT, filter->data));
 				}
 			} else { /* illegal character */
+				filter->cache = filter->status = 0;
 				CK((*filter->output_function)(MBFL_BAD_INPUT, filter->data));
 			}
-			filter->cache = filter->status = 0;
```

Comment:
I'm confused, didn't you move the common assignment out of the conditions in ext/mbstring/libmbfl/filters/mbfilter_utf16.c ? Why move them in here?

Reply:
We need to set `filter->cache` and `filter->status` before calling the `CK` macro, because `CK` can cause an abrupt return, which would leave the filter in an invalid state.

At the same time, I can't move them *before* the `if` tests which check `filter->status` and `filter->cache`, because then the tests wouldn't work as required.